internal/static/cartogram2.js

```diff
@@ -39,21 +39,55 @@ function clearFileInput(ctrl) {
     if (ctrl.value) {
       ctrl.parentNode.replaceChild(ctrl.cloneNode(true), ctrl);
     }
-  }
+ }
+
+function addClipboard (button_id, message) {
+
+    $("#" + button_id).tooltip({
+        trigger : 'hover',
+      })
+      
+    document.getElementById(button_id).onclick = function() {
+        var icon_id = button_id + "-icon";
+        navigator.clipboard.writeText(message);
+        document.getElementById(icon_id).src = 'static/clipboard-check.svg';
+        $("#" + button_id)
+        .attr('data-original-title', "Copied!")
+        .tooltip('show');
+        
+        setTimeout(function() {
+            document.getElementById(icon_id).src = 'static/clipboard.svg';
+            $("#" + button_id)
+            .attr('data-original-title', "Copy")
+        }
+        , 2000);
+    };
+}
+  
+function removeCCLogoFromMobileDevices() {
+    // Following code removes the creative commons logo if the website is opened from a mobile browser
```

Comment:
Why is the CC logo hidden on mobile browsers? Is there not enough screen space? If so, targeting the user agent isn't a good practice for doing this. Instead, use [CSS media queries](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#targeting_media_features) (see the `max-width` query on the linked page) to selectively hide the CC logo when the screen width is too small.

A cornerstone of responsive design is that we do not target specific devices, per se, but device _properties_ (like screen width) instead, which are more resilient.